core/unit_test/TestAtomics.hpp

```diff
@@ -122,7 +122,7 @@ struct SuperScalar {
   }
 
   KOKKOS_INLINE_FUNCTION
-  bool operator==(const SuperScalar& src) {
+  bool operator==(const SuperScalar& src) const {
```

Comment:
If we are going to change these anyway, make them hidden friends:

friend bool operator==(const SuperScalar& lhs, const SuperScalar& res) { /*... */ }

Also, while not technically part of this change, is there any reason (performance?) that operator!= doesn't just call operator==?

Reply:
This is just a test...